app/services/my_anime_list_sync_service.rb

```diff
@@ -0,0 +1,203 @@
+                status: format_status(le.status),
+                chapters: le.progress,
+                score: format_score(le.rating)
+              }
+            )
+          end
+        end
+      end
+
+      # if block is not executed because
+      # response.code was not a 200
+      # TODO: check timing
+      # return nil
+    end
+  end
+
+  def format_status(status)
+    # change our status -> mal status
+    case status
+    when 'current' then 1 # watching/reading
+    when 'planned' then 6 # plan to watch/plan to read
+    when 'completed' then 2 # completed
+    when 'on_hold' then 3 # on hold
+    when 'dropped' then 4 # dropped
+    end
+  end
+
+  # if you send no score in
+  # ie: &score&anythingelse
+  # it will not set the score
+  def format_score(score)
+    return nil if score.nil?
+
+    (score * 2).floor
+  end
+
+  private
+
+  def get(url, profile)
+    request = Typhoeus::Request.new(
+      build_url(url),
+      method: :get,
+      userpwd: simple_auth(profile)
+    )
+
+    request_status(request) do |response|
+      yield response
+    end
+    request.run
+  end
+
+  def post(url, profile, body)
+    request = Typhoeus::Request.new(
+      build_url(url),
+      method: :post,
+      userpwd: simple_auth(profile),
+      body: body
+    )
+
+    # TODO: @nuck is there a better way to handle this?
+    request_status(request) do |response|
+      # doesn't need to do anything if success
+    end
+    request.run
+  end
+
+  def put(url, profile, body)
+    request = Typhoeus::Request.new(
+      build_url(url),
+      method: :put,
+      userpwd: simple_auth(profile),
+      body: body
+    )
+
+    # TODO: @nuck is there a better way to handle this?
+    request_status(request) do |response|
+      # doesn't need to do anything if success
+    end
+    request.run
+  end
+
+  def delete(url, profile)
+    request = Typhoeus::Request.new(
+      build_url(url),
+      method: :delete,
+      userpwd: simple_auth(profile)
+    )
+
+    # TODO: @nuck is there a better way to handle this?
+    request_status(request) do |response|
+      # doesn't need to do anything if success
+    end
+    request.run
+  end
+
+  def request_status(request)
+    # will return request or nil
+    request.on_complete do |response|
+      if response.success?
+        # this is being sent up to either
+        # get/create/update/delete
+        # afterwards the chosen method will send it up
+        # to the parent request under my_anime_list.rb
+
+        # delete will not have a body so need try catch
+        yield response&.body
+      elsif response.timed_out?
+        # aw hell no
+        $stderr.puts('got a time out')
+      elsif response.code.zero?
+        # Could not get an http response, something's wrong.
+        $stderr.puts(response.return_message)
+      else
+        # Received a non-successful http response.
+        $stderr.puts('HTTP request failed: ' + response.code.to_s)
+      end
+    end
+  end
```

Comment:
It's probably better to raise an exception when we get a non-200 response so that it gets caught by Sentry and emailed to me ;)